tests/test_errors.py

```diff
@@ -116,15 +116,28 @@ def test_validate_enum_error(self):
             message == "Expected: a member of the enum ['Cat']\n\nReceived: \"Turtle\""
         )
 
+    def test_format_validation_passes_for_string_type_with_numeric_format(self):
+        d = [
+            ({"format": "double", "type": "string"}, "3.12"),
+            ({"format": "integer", "type": "string"}, "312"),
+            ({"format": "float", "type": "string"}, "3.12"),
+            ({"format": "number", "type": "string"}, "3.12"),
+        ]
+        for schema, data in d:
```

Comment:
We can use [@pytest.mark.parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html) here for each of the tested scenarios, this will isolate each of them as a single tests and have better test outputs. 

The same can be applied to the other existing tests, but for this one would be good to start :)

Reply:
@maticardenas Not sure if it's possible for this particular test file, as I am getting an error when trying to add it. I think this is because instead of being a standard test file, it's a class with tests (didn't even know it was possible and don't know why it's like this) and I think this is the reason it was written like this to begin with. So I think the file should be fully re-written to make this change, but I would do this as a separate PR

Reply:
I see, yes, indeed the tests are included within the class I assume for grouping reasons. 
Let's handle that in a separate PR, agreed. thanks